### PR TITLE
Adds sample weights to accuracy_score

### PR DESCRIPTION
**Pull request by jrbourbeau**
This PR adds implements sample weighting for `accuracy_score`

### CODE REVIEW

File: dask_ml/metrics/classification.py [hunk 1]
```diff
@@ -20,9 +24,8 @@ def accuracy_score(y_true, y_pred, normalize=True, sample_weight=None,
         If ``False``, return the number of correctly classified samples.
         Otherwise, return the fraction of correctly classified samples.
 
-    sample_weight : None
-        For compatibility with scikit-learn. This is currently not
-        supported.
+    sample_weight : 1d array-like, optional
+        Sample weights.
```
Comment (TomAugspurger) [inline dask_ml/metrics/classification.py]: Could you add a

```
    .. versionadded:: 0.7.0
```

directive below `Sample weights`.

We should start doing this for all new features / keywords.

**Comment by TomAugspurger** [comment]
Looks good overall.

Only thing might be how this behaves with dask < 0.18.0. If someone has dask 0.17, they'll get `AttributeError: module 'dask.array' has no attribute 'average'`. Maybe we could check the dask version and raise a nicer error message if it's too old?

**Comment by jrbourbeau** [comment]
Good point, I'll check the dask version number and raise a `NotImplementedError` accordingly

**Comment by jrbourbeau** [comment]
I thought checking the dask version might end up being useful elsewhere, so I added `DASK_VERSION` to `dask_ml._compat` (similar to the existing `SK_VERSION`).

**Comment by TomAugspurger** [comment]
Sorry, thought this was merged.

**Comment by jrbourbeau** [comment]
No probs, thanks @TomAugspurger!

**Pull request merged**